BOOTSEQ.md

```diff
@@ -4,77 +4,160 @@
   - With a pre-provisioned, local host OS present, and for other functions (most notably, networking), the host can continue its boot sequence. Ports should appear down from the host's perspective until the xPU is ready to handle traffic from/to host.
-  - Host OS continues to its boot. xPU needs to either respond to PCIe transactions or respond with retry config cycles(CRS) till it is ready. This automatically holds the Host BIOS. Note that Host BIOS may need to be modified to allow longer CRS response time before timing out.
-- Should xPU OS wait for host to finish boot ?
-  - How tightly coupled should host and xPU be during boot phase?
-- who tells Host OS/xPU OS to wait and how ?
-  - PCIe has a mechanism for config retry cycles(CRS) that can be used.
-- who tells Host OS to to continue / start booting ?
-  - These events may not be synchronous. The xPU may be installed in a PICe slot providing persistent power. xPU power cycle may be a function of the host BMC (and potentially the xPU BMC) rather than the power state of the host
-- is Host CPU halted ?
-- can we wait in UEFI/BIOS in case of network boot via xPU ?
+  - Host OS continues to its boot. xPU needs to either respond to PCIe transactions or respond with retry config cycles (CRS) till it is ready. This automatically holds the host BIOS.
+  - The xPU may hold its PCIe channel down during xPU boot. When the xPU has finished booting, it generates a hot-plug event, and the host will insert the xPU into the PCIe root complex. Is this true for all host OS?
+
+- Who tells host OS/xPU OS to wait and how ?
+  - PCIe has a mechanism for config retry cycles (CRS) that can be used
+  - Host BIOS may need to be modified to allow longer CRS response time before timing out
+
+- Host BIOS may sequence the start-up either automatically using orchestration built into BIOS or explicitly through commands sent by external orchestrator to BIOS
+  - Provide power to xPU without starting host
+  - Wait for xPU to complete startup
+  - Boot host
+
+- Who tells host OS to continue / start booting ?
+  - Default is for host OS continues boot process unless it is paused by CRS signal from xPU ?
+
+- Are there any circumstances under which host CPU is halted ?
+
+- Can we wait in UEFI/BIOS in case of network boot via xPU ?
   - Some UEFI PXE/HTTP boots have timeouts, sometimes painfully short ones (as low as 30 seconds)
-  - If the XPU provides an UNDI driver for network boot in its OPROM, it can have the Host BIOS wait.
+  - If the xPU provides an UNDI driver for network boot in its OPROM, it can have the host BIOS wait.
+
 - Should host continue booting even if xPU failed to boot after timeout ?
-  - Yes. XPU services/devices will be unavailable
+  - Yes. xPU services/devices will be unavailable
+  - Yes. May be needed to overwrite a corrupt image on the xPU
+
+- Should xPU OS wait for host to finish boot ?
+  - Only if use case requires the xPU to have dependencies on resources owned by the host ?
```

Comment:
xPUs are independent subsystem, they don't depend on any hard resources from the host (e.g. memory), then I don't see a need. Let's wait for others to chime in as well.

Reply:
I am not aware of any conditions in which the xPU is dependent on x86 resources.  Don't know if CXL will change this.  
xPU sharing x86 resources seems susceptible to race conditions and requires careful consideration.